pkg/client/cache/reflector.go

```diff
@@ -85,21 +85,28 @@ func NewReflector(lw ListerWatcher, expectedType interface{}, store Store, resyn
 // Run starts a watch and handles watch events. Will restart the watch if it is closed.
 // Run starts a goroutine and returns immediately.
 func (r *Reflector) Run() {
-	go util.Forever(func() { r.listAndWatch() }, r.period)
+	go util.Forever(func() { r.listAndWatch(neverStopWatch) }, r.period)
 }
 
 // RunUntil starts a watch and handles watch events. Will restart the watch if it is closed.
 // RunUntil starts a goroutine and returns immediately. It will exit when stopCh is closed.
 func (r *Reflector) RunUntil(stopCh <-chan struct{}) {
-	go util.Until(func() { r.listAndWatch() }, r.period, stopCh)
+	go util.Until(func() { r.listAndWatch(stopCh) }, r.period, stopCh)
 }
 
 var (
 	// nothing will ever be sent down this channel
 	neverExitWatch <-chan time.Time = make(chan time.Time)
 
+	// a dummy stop channel that never sends a stop signal
+	neverStopWatch <-chan struct{} = make(chan struct{})
```

Comment:
Reuse this one: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/util/util.go#L97